libraries/bot-azure/src/test/java/com/microsoft/bot/azure/TranscriptStoreTests.java

```diff
@@ -0,0 +1,589 @@
+                Assert.assertEquals("echo:foo", pagedResult.getItems().get(2).getText());
+            } else {
+                Assert.fail();
+            }
+
+            if(pagedResult.getItems().get(3).isType(ActivityTypes.MESSAGE)) {
+                Assert.assertEquals("bar", pagedResult.getItems().get(3).getText());
+            } else {
+                Assert.fail();
+            }
+
+            Assert.assertTrue(pagedResult.getItems().get(4) != null);
+            if(!pagedResult.getItems().get(4).isType(ActivityTypes.TYPING)) {
+                Assert.fail();
+            }
+
+            if(pagedResult.getItems().get(5).isType(ActivityTypes.MESSAGE)) {
+                Assert.assertEquals("echo:bar", pagedResult.getItems().get(3).getText());
+            } else {
+                Assert.fail();
+            }
+
+            for (Activity activity: pagedResult.getItems()) {
+                Assert.assertTrue(!StringUtils.isBlank(activity.getId()));
+                //this is the default DateTimeOffset in C#
+                OffsetDateTime defaultDate = OffsetDateTime.of(0001,1,1,12,0,0,0,OffsetDateTime.now().getOffset());
+                Assert.assertTrue(activity.getTimestamp().isAfter(defaultDate));
+            }
+        }
+    }
+
+    @Test
+    public void logUpdateActivities() {
+        if (EMULATOR_IS_RUNNING) {
+            ConversationReference conversation = TestAdapter
+                .createConversationReference(UUID.randomUUID().toString(), "User1", "Bot");
+            TestAdapter adapter = new TestAdapter(conversation).use(new TranscriptLoggerMiddleware(transcriptStore));
+            final Activity[] activityToUpdate = {null};
+            new TestFlow(adapter, turnContext -> {
+                activityToUpdate[0] = new Activity(ActivityTypes.MESSAGE);
+                if(turnContext.getActivity().getText().equals("update")) {
+                    activityToUpdate[0].setText("new response");
+                    turnContext.updateActivity(activityToUpdate[0]).join();
+                } else {
+                    Activity activity = turnContext.getActivity().createReply("response");
+                    ResourceResponse response = turnContext.sendActivity(activity).join();
+                    activity.setId(response.getId());
+
+                    // clone the activity, so we can use it to do an update
+                    JacksonAdapter jacksonAdapter = new JacksonAdapter();
+                    try {
+                        activityToUpdate[0] = jacksonAdapter.deserialize(jacksonAdapter.serialize(activity), Activity.class);
+                    } catch (IOException exception) {
+                        exception.printStackTrace();
+                    }
+                }
+                return CompletableFuture.completedFuture(null);
+            }).send("foo")
+                .send("update")
+                .assertReply("new response")
+                .startTest().join();
+        }
```

Comment:
Are you missing `asserts`? See [C#](https://github.com/microsoft/botbuilder-dotnet/blob/62bdb513fe628f89924567448946b2289b8fc203/tests/Microsoft.Bot.Builder.Azure.Tests/TranscriptStoreBaseTests.cs#L285)